Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -1890,7 +1896,7 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseDebuggerState
 }
 
 template <typename LexerType>
-template <class TreeBuilder> TreeStatement Parser<LexerType>::parseBlockStatement(TreeBuilder& context, bool isCatchBlock)
+template <class TreeBuilder> TreeStatement Parser<LexerType>::parseBlockStatement(TreeBuilder& context, bool isCatchBlock, bool isStaticBlock)
```

Comment:
Instead of passing two boolean, we should pass one enum which indicates the block type, NormalBlock, CatchBlock, and ClassStaticBlock.